numpy/lib/function_base.py

```diff
@@ -4839,6 +4843,22 @@ def digitize(x, bins, right=False):
     if mono == 0:
         raise ValueError("bins must be monotonically increasing or decreasing")
 
+    if edge:
+        # =========  =============  ============================ ===== =====
+        # `right`    order of bins  returned index `i` satisfies delta index
+        # =========  =============  ============================ ===== =====
+        # ``False``  increasing     ``bins[i-1] <= x < bins[i]``   1    -1
+        # ``True``   increasing     ``bins[i-1] < x <= bins[i]``   -1    0
+        # ``False``  decreasing     ``bins[i-1] > x >= bins[i]``   1    0
+        # ``True``   decreasing     ``bins[i-1] >= x > bins[i]``   -1    -1
+        # =========  =============  ============================ ===== =====
+        delta = -1 if right else 1
+        idx = -1 if delta == mono else 0
+        if np.issubdtype(bins.dtype, _nx.integer):
+            bins[idx] += delta
```

Comment:
This code is vulnerable to overflow and behaves incorrectly:
```python
# ok
>>> digitize([0, 1, 2, 3], np.array([0, 2, 255], dtype=np.uint8))
array([1, 1, 2, 2], dtype=int64)

# wrong!
>>> digitize([0, 1, 2, 3], np.array([0, 2, 255], dtype=np.uint8), edge=True)
array([1, 3, 3, 3], dtype=int64)
```

Reply:
I think you needed to use the suggestion from [this comment](https://github.com/numpy/numpy/pull/16248#issuecomment-629415767)

Reply:
Ok working on fixing

Reply:
Good catch